controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -362,6 +371,30 @@ func (r *Reconciler) ReconcileNormal(ctx *pkgctx.VolumeContext) error {
 		volumeStatusesForLegacy,
 	)
 
+	if len(beforeStatusVolumes) > 0 {
+		for _, vol := range ctx.VM.Status.Volumes {
+			if vol.Type != vmopv1.VolumeTypeManaged {
+				continue
+			}
+
+			name := strings.TrimSuffix(vol.Name, volumeNameDetachSuffix)
+			// Might not know the UUID before being attached, but also the volume
```

Comment:
Do you want to mention this in the logging as well? The list might not be complete.

Reply:
Yah it might be nice to copy the PVC name into the Status so we have that to log instead of just the uuid but I'll leave if we want to do that in a separate change.